### PR TITLE
Fix Inkling inputs_embeds and add more tests

### PR DESCRIPTION
**Pull request by Cyrilvallez**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47827) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47827)

# What does this PR do?

As per the title. Since `get_input_embedding` does not return the norm module (they are 2 separate modules), we need to apply the `norm` outside the `if`, since inputs_embeds will not be normed if passed explicitly in `forward` instead of `input_ids`

### CODE REVIEW

File: src/transformers/models/inkling/modeling_inkling.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def __init__(self, config):
 
         self.weight = nn.Parameter(torch.empty(self.n_total_experts, config.hidden_size))
         self.global_scale = nn.Parameter(torch.ones(1))
-        self.e_score_correction_bias = nn.Parameter(torch.empty(self.num_experts))
+        self.e_score_correction_bias = nn.Buffer(torch.empty(self.num_experts))
```
Comment (Cyrilvallez) [inline src/transformers/models/inkling/modeling_inkling.py]: It's a Buffer on every other MoE. Moreover, having a Parameter here with an `_init` of `zeros` means that it will NEVER be trained as it's only use as an addition, and adding 0 will never create gradients flowing back to it

**Comment by ydshieh** [comment]
Please use run-slow sir !
